core/state/journal.go

```diff
@@ -153,34 +169,30 @@ type (
 
 func (ch createObjectChange) revert(s *StateDB) {
 	delete(s.stateObjects, *ch.account)
-	delete(s.stateObjectsDirty, *ch.account)
 }
 
 func (ch createObjectChange) dirtied() *common.Address {
 	return ch.account
 }
 
-func (ch resetObjectChange) revert(s *StateDB) {
-	s.setStateObject(ch.prev)
-	if !ch.prevdestruct {
-		delete(s.stateObjectsDestruct, ch.prev.address)
-	}
-	if ch.prevAccount != nil {
-		s.accounts[ch.prev.addrHash] = ch.prevAccount
-	}
-	if ch.prevStorage != nil {
-		s.storages[ch.prev.addrHash] = ch.prevStorage
-	}
-	if ch.prevAccountOriginExist {
-		s.accountsOrigin[ch.prev.address] = ch.prevAccountOrigin
-	}
-	if ch.prevStorageOrigin != nil {
-		s.storagesOrigin[ch.prev.address] = ch.prevStorageOrigin
+func (ch createObjectChange) copy() journalEntry {
+	return createObjectChange{
+		account: ch.account,
 	}
 }
 
-func (ch resetObjectChange) dirtied() *common.Address {
-	return ch.account
+func (ch createContractChange) revert(s *StateDB) {
+	s.stateObjects[ch.account].created = false
```

Comment:
Given that all other journal entries use `s.getStateObject(*ch.account).SetXXX` when they revert; and also createContractChange is constructed as below:

```go
	obj := s.getStateObject(addr)
	if !obj.created {
		obj.created = true
		s.journal.append(createContractChange{account: addr})
	}
```

Shouldn't we use `s.getStateObject(ch.account).created = false` here instead?

Reply:
I mean, your code is fine too, just thinking about consistency across the journal entries.